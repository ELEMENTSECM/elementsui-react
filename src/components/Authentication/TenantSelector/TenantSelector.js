import * as React from 'react';
import PropTypes from 'prop-types';
import Button from '../../Inputs/Button';
import Dropdown from '../../Inputs/Dropdown';
import Spinner from '../../Indicators/Spinner';
import { Link } from 'office-ui-fabric-react/lib/Link';
import { MessageBar, MessageBarType } from 'office-ui-fabric-react/lib/MessageBar';

/** TenantSelector example */
class TenantSelector extends React.Component {
	static defaultProps = {
		labels: {
			login: 'Log in',
			logout: 'Log out',
			selectTenant: 'Select tenant',
			loggedInAs: 'You are logged in as'
		}
	};

	constructor(props) {
		super(props);

		this.state = {
			selectedDatabase: null
		};
	}

	callOnChange = value => {
		if (this.props.tenants != null) {
			this.props.tenants.forEach(tenant => {
				if (tenant._id === value) {
					this.props.onChange(tenant);
				}
			});
		}
	};

	handleSelectChange = option => {
		this.callOnChange(option.key);
		this.setState({
			selectedDatabase: option.key
		});
	};

	render() {
		let options;
		if (this.props.tenants != null && !this.props.isLoggedIn) {
			options = this.props.tenants.map(tenant => ({
				key: tenant._id,
				text: tenant._id,
				isSelected: tenant._id === this.props.preSelectedDataBase
			}));
		}

		return (
			<div>
				{!this.props.isLoggedIn ? (
					<div>
						<Dropdown
							id="databaseDropdown"
							label={this.props.labels.selectTenant + ':'}
							selectedKey={this.state.selectedDatabase}
							placeHolder={this.props.labels.selectTenant + '...'}
							options={options}
							onChange={option => this.handleSelectChange(option)}
						/>
						<br />
						<Button
							label={this.props.labels.login}
							disabled={this.state.selectedDatabase == null}
							onClick={() => this.props.handleLoginClick()}
						/>
						<div>{this.props.isSpinnerVisible && <Spinner />}</div>
					</div>
				) : (
					<MessageBar messageBarType={MessageBarType.info}>
						{`${this.props.labels.loggedInAs} '${this.props.name}'. `}
						<Link href="" onClick={this.props.handleLogoutClick}>
							{this.props.labels.logout}
						</Link>
					</MessageBar>
				)}
			</div>
		);
	}
}

TenantSelector.propTypes = {
	/** List of tenant configs */
	tenants: PropTypes.arrayOf(
		PropTypes.shape({
			_id: PropTypes.string,
			_childId: PropTypes.string,
			_scope: PropTypes.string,
			ncoreclient: PropTypes.shape({
				BaseUrl: PropTypes.string
			}),
			elements: PropTypes.shape({
				Authentication_BaseUrl: PropTypes.string,
				Authentication_DefaultProvider: PropTypes.string
			})
		})
	),
	/** Logged in user's name */
	name: PropTypes.string,
	/** Is logged in */
	isLoggedIn: PropTypes.bool,
	/** Selected tenant changed event handler */
	onChange: PropTypes.func,
	/** Log in mouse click event handler */
	handleLoginClick: PropTypes.func,
	/** Log ou mouse click event handler */
	handleLogoutClick: PropTypes.func,
	/** Is spinner visible */
	isSpinnerVisible: PropTypes.bool,
	/** Labels */
	labels: PropTypes.shape({
		/** Log in button label */
		login: PropTypes.string,
		/** Log out button label */
		logout: PropTypes.string,
		/** Select tenant label */
		selectTenant: PropTypes.string,
		/** Logged in label */
		loggedInAs: PropTypes.string
	})
};

export default TenantSelector;
