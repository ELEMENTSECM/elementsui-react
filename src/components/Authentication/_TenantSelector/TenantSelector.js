import * as React from 'react';
import PropTypes from 'prop-types';
import Dropdown from '../../Inputs/Dropdown';
import LoggedInBar from '../_LoggedInBar';
import Label from '../../Inputs/Label';
import { styles } from './TenantSelector.styles';
import { classNamesFunction, customizable, styled } from 'office-ui-fabric-react/lib/Utilities';
import { FormattedMessage } from 'react-intl';
/** TenantSelector example */
class TenantSelector extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			selectedDatabase: this.props.selectedTenant
		};

		this.classNames = classNamesFunction()(styles, props);
	}

	callOnChange = value => {
		if (this.props.tenants != null) {
			this.props.tenants.forEach(tenant => {
				if (tenant.Id === value) {
					this.props.onChange(tenant.Id);
				}
			});
		}
	};

	handleSelectChange = option => {
		this.callOnChange(option.key);
		this.setState(ps => ({
			...ps,
			selectedDatabase: option.key
		}));
	};

	componentWillReceiveProps(newProps) {
		this.setState(ps => ({ ...ps, selectedDatabase: newProps.selectedTenant }));
	}

	render() {
		let options;
		if (this.props.tenants != null && !this.props.isLoggedIn) {
			options = this.props.tenants.map(tenant => ({
				key: tenant.Id,
				text: tenant.Description,
				isSelected: tenant.Id === this.props.selectedTenant
			}));
		}

		return (
			<div>
				{!this.props.isLoggedIn ? (
					<div>
						<Label className={this.classNames.label}>
							<FormattedMessage id="database" />
						</Label>
						<Dropdown
							id="databaseDropdown"
							className={this.classNames.selectTenantDropdown}
							selectedKey={this.state.selectedDatabase}
							placeHolder={'...'}
							options={options || []}
							onChange={option => this.handleSelectChange(option)}
						/>
					</div>
				) : (
					<LoggedInBar
						currentUserName={this.props.currentUserName}
						tenant={this.state.selectedDatabase}
						handleLogoutClick={this.props.handleLogoutClick}
						goBack={this.props.goBack}
						logout={this.props.logout}
					/>
				)}
			</div>
		);
	}
}

TenantSelector.propTypes = {
	/** Selected tenant */
	selectedTenant: PropTypes.string,
	/** List of tenant configs */
	tenants: PropTypes.arrayOf(
		PropTypes.shape({
			Id: PropTypes.string,
			Description: PropTypes.string
		})
	),
	/** Logged in user's name */
	currentUserName: PropTypes.string,
	/** Is logged in */
	isLoggedIn: PropTypes.bool,
	/** Selected tenant changed event handler */
	onChange: PropTypes.func,
	/** Back button click event handler */
	goBack: PropTypes.func,
	/** User-defined styling */
	styles: PropTypes.func
};

export default styled(customizable('TenantSelector', ['theme'])(TenantSelector), styles);
