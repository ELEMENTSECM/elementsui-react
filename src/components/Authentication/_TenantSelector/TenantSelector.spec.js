import React from 'react';
import renderer from 'react-test-renderer';
import TenantSelector from './TenantSelector';
import { IntlProvider } from 'react-intl';
import json from '../Login/Login.nls.json';

describe('TenantSelector', () => {
	const tenants = [{ Id: 'ID_1', Description: 'ID_1_DESC' }];

	test('displays tenants and labels correctly', () => {
		const tree = renderer
			.create(
				<IntlProvider locale="en" messages={json.en}>
					<TenantSelector tenants={tenants} selectedTenant="ID_1" currentUserName="John Doe" />
				</IntlProvider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('displays when logged in correctly', () => {
		const tree = renderer
			.create(
				<IntlProvider locale="en" messages={json.en}>
					<TenantSelector
						tenants={tenants}
						isLoggedIn={true}
						selectedTenant="ID_1"
						currentUserName="John Doe"
					/>
				</IntlProvider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
