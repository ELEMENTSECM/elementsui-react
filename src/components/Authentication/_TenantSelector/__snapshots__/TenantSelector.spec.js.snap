// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TenantSelector displays tenants and labels correctly 1`] = `
<div>
  <div>
    <div
      className="ms-Dropdown-container"
    >
      <label
        className="ms-Label ms-Dropdown-label root-41"
        htmlFor="databaseDropdown"
        id="databaseDropdown-label"
        required={undefined}
      >
        Select database:
      </label>
      <div
        aria-activedescendant={undefined}
        aria-describedby="databaseDropdown-option"
        aria-disabled={undefined}
        aria-expanded="false"
        aria-label={undefined}
        aria-live="assertive"
        aria-owns={undefined}
        className="ms-Dropdown root_aecb6435 ms-Dropdown root-40"
        data-is-focusable={true}
        id="databaseDropdown"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <span
          aria-atomic={true}
          className="ms-Dropdown-title title_aecb6435 ms-Dropdown-titleIsPlaceHolder titleIsPlaceHolder_aecb6435"
          id="databaseDropdown-option"
          role="listbox"
        >
          <span>
            Select database...
          </span>
        </span>
        <span
          className="ms-Dropdown-caretDownWrapper caretDownWrapper_aecb6435"
        >
          <i
            aria-hidden={true}
            className="ms-Dropdown-caretDown caretDown_aecb6435 root-42"
            data-icon-name="ChevronDown"
            role="presentation"
          />
        </span>
      </div>
    </div>
    <br />
    <button
      aria-describedby={undefined}
      aria-label={undefined}
      aria-labelledby={undefined}
      aria-pressed={undefined}
      className="ms-Button ms-Button--default is-disabled root-43"
      data-is-focusable={false}
      disabled={true}
      id={undefined}
      onClick={[Function]}
      type="button"
    >
      <div
        className="ms-Button-flexContainer flexContainer-44"
      >
        <div
          className="ms-Button-textContainer textContainer-45"
        >
          <div
            className="ms-Button-label label-47"
            id="id__1"
          >
            Sign in
          </div>
        </div>
      </div>
    </button>
    <div />
  </div>
</div>
`;

exports[`TenantSelector displays when logged in correctly 1`] = `
<div>
  <div
    aria-live="polite"
    className="ms-MessageBar root_613660a4 ms-MessageBar root_613660a4 ms-MessageBar-multiline multiLine_613660a4"
    role="status"
  >
    <div
      className="content_613660a4 ms-MessageBar-content"
    >
      <div
        className="ms-MessageBar-icon icon_613660a4"
      >
        <i
          aria-hidden={true}
          className="root-42"
          data-icon-name="Info"
          role="presentation"
        />
      </div>
      <div
        className="ms-MessageBar-text text_613660a4 noDismissButton_613660a4"
        id="MessageBar4"
      >
        <span
          aria-live="polite"
          className="ms-MessageBar-innerText innerText_613660a4"
          role="status"
        />
      </div>
    </div>
  </div>
</div>
`;
